rocketchat-core/src/main/java/com/rocketchat/core/ChatRoom.java

```diff
@@ -15,25 +16,41 @@
 import com.rocketchat.core.model.Message;
 import com.rocketchat.core.model.RoomRole;
 import com.rocketchat.core.model.Subscription;
+import com.rocketchat.core.roomstream.Document.FileDocument;
+import com.rocketchat.core.roomstream.Document.MessageDocument;
+import com.rocketchat.core.roomstream.LocalStreamCollectionManager;
 import com.rocketchat.core.uploader.FileUploader;
-
-import org.json.JSONObject;
-
 import java.util.Date;
+import org.json.JSONObject;
 
 // TODO: 29/7/17 add throw custom exceptions if method call violates permission required to execute given RPC
 public class ChatRoom {
 
     private final RocketChatClient client;
     private final BaseRoom room;
+    private LocalStreamCollectionManager localStreamCollectionManager;
 
     //Subscription Ids for new subscriptions
-    private String roomSubId;  // TODO: 29/7/17 check for persistent SubscriptionId of the room
+    private String roomSubId;
     private String typingSubId;
+    private String deleteSubId;
+
+
+    //subscription Ids for room collections
+    private String filesSubId;
+    private String mentionedMessagesSubId;
+    private String starredMessagesSubId;
+    private String pinnedMessagesSubId;
+    private String snipetedMessagesSubId;
 
     public ChatRoom(RocketChatClient api, BaseRoom room) {
         this.client = api;
         this.room = room;
+        this.localStreamCollectionManager = new LocalStreamCollectionManager();
```

Comment:
This probably should be injected in the constructor, so you can use a Mock version on tests....